rfc/007-monitoring.org

```diff
@@ -0,0 +1,231 @@
+
+** setBoolArrayAttribute
+
+A function that allow to add a boolean array attribute to the current started span.
+
+** addEvent
+
+A function that allow to add an Event to the current started span.
+
+** trace,  debug, info, warn, error and fatal
+
+Functions that allow to add an Log Event with respectively (Trace, Debug, Info, Warn, Error and Fatal) level to the current started span using Logger mechanism.
+
+** setOkStatus
+
+A function that set the status of current started span to Ok.
+
+** setErrorStatus
+
+A function that set the status of current started span to Error with the corresponding message.
+
+** setUnsetStatus
+
+A function that set the status of current started span to default Unset value.
+
+** endSpan
+
+A function that end the current started span, flush the resulting message and clean the use memory.
+If the span have a parent, the parent become the current span.
+
+* Properties
+
+** SKMonitor
+
+Each root span defines the following attributes:
+
+- skruntime.memory.persistent: The persistent memory size used after the run (in Bytes).
+- skruntime.memory.free: The size of the free table remaining after the run (in Bytes).
+- skruntime.obstack.pick: The maximun obstack size used during the run (in Bytes).
+
+Each span defines the following attributes:
+
+- skmonitor.memory: The memory used by the span itself.
+
+** SKDB
+
+SKDB will creates at least on span for each command.
+
+For root span the name of the span represent the current executed command.
+
+During SKDB execution the following attributes may be generated:
+
+- skdb.database: The name of the current executed database (if --init or --data supplied).
+- skdb.bytes.read: The number of bytes read durring the command.
+- skdb.bytes.written: The number of bytes written durring the command.
+- skdb.replication.reads: The number of replication lines read durring the command.
+- skdb.replication.writes: The number of replication lines written durring the command.
+
+
+During SKDB `exec` command a span will be generated for each transaction and one for each statement.
+For transaction spans the span name is "skdb.transaction".
+For statement spans the span name is "skdb.statement".
+
+Durring statement execution the following attributes may be generated:
+
+- skdb.stmt.type: The type of the SQL Statemement.
+
+Other properties could be added to have more info on executed statement sush as:
+
+- The number of columns.
+- The types of columns.
+- The number of joins.
+- The number of tables implied.
+- ...
+
+* IO
+
+** Writer
+
+By default the monitoring trace is written to stderr, a SK_MONITORING_WRITER environment variable can be defined with following value:
+
+- 'fifo' to write the trace into '/tmp/skmonitor.trace' fifo.
+- 'file' to write the trace into 'skmonitor.log' file.
+- 'fifo:<name>' to write the trace into '<name>' fifo.
+- 'file:<name>' to write the trace into '<name>' file.
+
+** Collector (TO BE DONE)
```

Comment:
If we're producing to the open telemetry spec, there must be stuff out there that can do this for us?